File: Stage01-Player/Lua/player.lua
local Player = {}
local Shared = require("shared")
--static player class - no constructor required
Player.Name 		= ""
Player.Health 		= 100
Player.Strength 	= 100
Player.Character 	= ""
Player.Characters 	= {}
Player.Inventory 	= {}

function Player.AddToInventory(item)
	--[[ add an item to player inventory ]]
end

function Player.DisplayInventory()
	--[[ display player's inventory ]]
end

function Player.DisplayPlayer()
	--[[ main use for debug. prints all player properties ]]
	print(string.rep('═', 60))
	print("Player properties:")
	print(string.rep('═', 60))
	print("Characters available: ".. table.concat(Player.Characters, ", "))
	print(string.rep('═', 60))
	print("Name:                 "..Player.Name)
	print("Health:               "..Player.Health)
	print("Strength:             "..Player.Strength)
	print("Character:            "..Player.Character)
	print("Inventory:            "..table.concat(Player.Inventory, ", "))
	print(string.rep('═', 60))
end

function Player.GetProperty(propertyName)
	--[[  used by narrator class to get a player property  ]]
	if propertyName:find("name") ~= nil then
		return Player.Name
	elseif propertyName:find("health") ~= nil then
		return Player.Health
	elseif propertyName:find("strength") ~= nil then
		return Player.Strength
	elseif propertyName:find("character") ~= nil then
		return Player.Character
	end
end

function Player.RemoveFromInventory(item)
	--[[  remove an item from player inventory ]]
end

function Player.UpdateStats(characterIndex)
	--[[ modify health and strength depending on character selected ]]
	Player.Health = Player.Health +  characterIndex * 2
	Player.Strength = Player.Strength - characterIndex * 2
end

return Player

File: Stage03-Locations/C#/Player.cs
using System;
using System.Collections.Generic;

namespace Adventure_03_World
{
    internal static class Player
    {
        public static string Name = "";
        public static int Health = 100;
        public static int Strength = 100;
        public static string Character = "";
        public static string ItemInHand = "";
        public static List<string> Characters = new List<string>();
        public static List<string> Inventory = new List<string>();

        public static void AddToInventory()
        {
            /// add an item to player inventory ///
        }
        public static void DisplayInventory()
        {
            /// display player's inventory ///
            Console.WriteLine(new string('═', Console.WindowWidth - 1));
            Console.WriteLine("Player properties:");
            Console.WriteLine(new string('═', Console.WindowWidth - 1));
            Console.Write("Characters available: ");
            foreach(string character in Characters)
                Console.Write($"{character}, ");
            Console.WriteLine($"\n{new string('═', Console.WindowWidth - 1)}");
            Console.WriteLine($"Name:                 {Name}");
            Console.WriteLine($"Health:               {Health}");
            Console.WriteLine($"Strength:             {Strength}");
            Console.WriteLine($"Character:            {Character}");
            Console.Write($"Inventory:            ");
            foreach (string item in Inventory)
                Console.Write($"{item}, ");
            Console.WriteLine(new string('═', Console.WindowWidth - 1));
        }
        public static void DisplayPlayer()
        {
            /// main use for debug. prints all player properties ///
            Console.WriteLine(new string('═', Console.WindowWidth - 1));
            Console.WriteLine("Player properties:");
            Console.WriteLine(new string('═', Console.WindowWidth - 1));
            Console.Write("Characters available: ");
            foreach (string character in Characters)
                Console.Write($"{character}, ");
            Console.WriteLine($"\n{new string('═', Console.WindowWidth - 1)}");
            Console.WriteLine($"Name:                 {Name}");
            Console.WriteLine($"Health:               {Health}");
            Console.WriteLine($"Strength:             {Strength}");
            Console.WriteLine($"Character:            {Character}");
            Console.Write($"Inventory:            ");
            foreach (string item in Inventory)
                Console.Write($"{item}, ");
            Console.WriteLine($"\n{new string('═', Console.WindowWidth - 1)}");
        }
        public static string GetProperty(string propertyName)
        {
            ///  used by narrator class to get a player properties ///
            if (propertyName.Contains("name"))
                return Name;
            else if (propertyName.Contains("health"))
                return Health.ToString();
            else if (propertyName.Contains("strength"))
                return Strength.ToString();
            else if (propertyName.Contains("character"))
                return Character;
            return "";
        }
        public static void RemoveFromInventory(string item)
        {
            ///  remove an item from player inventory ///
        }
        public static void UpdateStats(int characterIndex)
        {
            ///  modify health and strength depending on character selected ///
            Health += characterIndex * 2;
            Strength += characterIndex * 2;
        }
    }
}
